src/encoded/static/components/motifs.js

```diff
@@ -0,0 +1,144 @@
+                this.sequenceLogos = require('d3-sequence-logo'); // logos This is for local development when changes are needed to d3-sequence-logo.
+                const targetElement = this.chartdisplay;
+                this.drawMotifs(targetElement);
+            }
+        });
+    }
+
+    // Redraw charts when window changes
+    componentDidUpdate() {
+        if (this.chartdisplay) {
+            this.chartdisplay.innerHTML = '';
+            const targetElement = this.chartdisplay;
+            this.drawMotifs(targetElement);
+        }
+    }
+
+    addMotifElement(targetElement, response, d, entryPoint) {
+        // Convert PWM text data to object
+        const PWM = convertTextToObj(response);
+        // Append information about the PWM including Biosample, Organ, and Targets
+        targetElement.insertAdjacentHTML('afterbegin', `<div class="motif-element" id="motif${d.accession}"></div>`);
+        const targetList = d.targets.map(t => t.label);
+        const targetListLabel = targetList.length > 1 ? 'Targets' : 'Target';
+        const htmlStr = `
+            <div class='motif-description'>
+                <p><a href=${d['@id']}>${d.accession}</a></p>
+                ${d.biosample_term_name ? `<p>Biosample: ${d.biosample_term_name}</p>` : ''}
+                ${d.organ_slims ? `<p>Organ: ${d.organ_slims.join(', ')}</p>` : ''}
+                ${targetList ? `<p>${targetListLabel}: ${targetList.join(', ')}</p>` : ''}
+            </div>`;
+        targetElement.insertAdjacentHTML('afterbegin', htmlStr);
+        // Generate the logo from the PWM object
+        entryPoint(`#motif${d.accession}`, PWM, this.d3);
+    }
+
+    drawMotifs(targetElement) {
+        const results = this.props.context['@graph'];
+        const urlBase = this.props.urlBase;
+
+        // Keep track of whether or not a PWM document has been identified
+        let emptyFlag = 1;
+
+        const entryPoint = this.sequenceLogos.entryPoint;
+        const d3 = this.d3;
+
+        let pwmLink = '';
+        // Check each element of @graph to see if it has documents and if the available documents are of the type "position weight matrix"
+        // If so, fetch PWM data and insert logo and associated PWM information to a list
+        results.forEach((d) => {
+            if (d.documents[0]) {
+                emptyFlag = 0;
+                if (d.documents[0].document_type === 'position weight matrix') {
+                    pwmLink = urlBase + d.documents[0]['@id'] + d.documents[0].attachment.href;
+
+                    getMotifData(pwmLink, this.context.fetch, entryPoint, d3).then((response) => {
+                        targetElement.insertAdjacentHTML('afterbegin', `<div class="element" id="element${d.accession}"></div>`);
+                        const targetElement2 = document.getElementById(`element${d.accession}`);
+                        this.addMotifElement(targetElement2, response, d, entryPoint, d3);
```

Comment:
addMotifElement doesn’t have a last d3 parameter — it uses this.d3.

Reply:
Fixed.